ambry-store/src/main/java/com/github/ambry/store/HybridCompactionPolicy.java

```diff
@@ -0,0 +1,192 @@
+/**
+ * Copyright 2020 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.store;
+
+import com.github.ambry.config.StoreConfig;
+import com.github.ambry.utils.Time;
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.codehaus.jackson.map.ObjectMapper;
+
+import static com.github.ambry.utils.Utils.*;
+
+
+/**
+ * HybridCompactionPolicy will run compaction with StatsBasedCompactionPolicy more frequently and with CompactAllPolicy
+ * one time out of storeCompactionPolicySwitchPeriod.
+ */
+public class HybridCompactionPolicy implements CompactionPolicy {
+  private final Time time;
+  private final StoreConfig storeConfig;
+  private static final Logger logger = LoggerFactory.getLogger(HybridCompactionPolicy.class);
+  private final Map<String, CompactionPolicySwitchInfo> blobToCompactionPolicySwitchInfoMap;
+  private final ObjectMapper mapper = new ObjectMapper();
+  private static final String COMPACTION_POLICY_COUNTER = "compactionPolicyCounter";
+  private static final String VALUE = "value";
+  private static final String LAST_COMPACT_ALL_TIME = "lastCompactAllTime";
+  private static final String COMPACT_POLICY_INFO_PATH = "/compactionPolicyInfo.";
```

Comment:
you can omit leading slash for two reasons
1. It's less cross-platform compatible, we can always use File.separator. Not that it matters here, just a nice pattern.
2. Use Paths.get is probably a better way to concatenate different parts of the file path.

Also, dataDir already has the partition id/replica id/store id as the last part of the filepath, we don't need to suffix storeId in the filename here. But we can add .json as extension, just to indicate this is a json formatted file.

Reply:
Updated.